### ISSUE TITLE
Phantasma Link Gas Issue

### ISSUE THREAD

**Issue by Relfos** [phantasma-io/TOMB#14]
As reported by Iggy:

I wanted to customize the minimum fee and gas limit to process big Tx.
This was the thing I had at first.
_let gasPrice = 100000 // changed
let gaslimit = 100000 // changed
let script = sb
    .callContract("GFEST", "mint", [myAddress, 1, 1])
    .callContract("GFEST", "mint", [myAddress, 2, 1])
    .callContract("GFEST", "mint", [myAddress, 2, 1])
    .callContract('gas', 'AllowGas', [myAddress, sb.nullAddress(), gasPrice, gaslimit]) // doesn’t recognize
    .callContract('gas', 'SpendGas', [myAddress])
    .endScript();_

This didn’t recognize the customized gasPrice and gaslimit value.
And this is the thing I have now and it works.
_let script = sb
    .callContract('gas', 'AllowGas', [myAddress, sb.nullAddress(), gasPrice, gaslimit])
    .callContract("GFEST", "mint", [myAddress, 1, 1])
    .callContract("GFEST", "mint", [myAddress, 2, 1])
    .callContract("GFEST", "mint", [myAddress, 2, 1])
    .callContract('gas', 'SpendGas', [myAddress])
    .endScript();_

**Comment by supportactivator** [comment]
Kindly open this live chat link to talk to the customer service directly on your issue https://direct.lc.chat/14234769

**Comment by Relfos** [comment]
This seems to be Javascript, not a TOMB issue, @TeknoPT please check, and tell me if it is ok to close this or if we should move it to another repo.

**Comment by TeknoPT** [comment]
Based on the explanation provided, the order of the script execution is crucial in determining the allowed gas and its subsequent utilization for the transactions. In the context of blockchain operations, gas refers to the computational resources required to execute a transaction, and thus, specifying the gas limit and gas price is essential for a smooth process.

In the first script you shared, the 'AllowGas' function is called at the end of the script. This means that the transactions for minting tokens would have already been executed before the gas limit and gas price are specified. As a result, the custom gasPrice and gaslimit values are not recognized, and the transactions may fail or revert due to insufficient gas allocation.

In the second script, the 'AllowGas' function is called at the beginning. This ensures that the gas limit and gas price are specified before executing the minting transactions. By specifying the gas allocation upfront, the script can effectively utilize the custom gasPrice and gaslimit values for the subsequent transactions.

In summary, when building a script to call the blockchain, it is important to ensure that the 'AllowGas' function is called before any other transactions, as it specifies the amount of gas allowed for the subsequent operations. Calling 'AllowGas' at the end does not allow the script to recognize the custom gas values, resulting in possible transaction failures due to insufficient gas allocation.
